kymatio/scattering2d/frontend/base_frontend.py

```diff
@@ -0,0 +1,107 @@
+        Convolutions are efficiently performed in the Fourier domain.
+
+        Example
+        -------
+            # 1) Define a Scattering2D object as:
+                s = Scattering2D_torch(J, shape=(M, N))
+            #    where (M, N) is the image size and 2**J the scale of the scattering
+            # 2) Forward on an input Tensor x of shape B x M x N,
+            #     where B is the batch size.
+                result_s = s(x)
+
+        Parameters
+        ----------
+        J : int
+            Log-2 of the scattering scale.
+        shape : tuple of ints
+            Spatial support (M, N) of the input.
+        L : int, optional
+            Number of angles used for the wavelet transform. Defaults to `8`.
+        max_order : int, optional
+            The maximum order of scattering coefficients to compute. Must be either
+            `1` or `2`. Defaults to `2`.
+        pre_pad : boolean, optional
+            Controls the padding: if set to False, a symmetric padding is applied
+            on the signal. If set to True, the software will assume the signal was
+            padded externally. Defaults to `False`.
+        backend : object, optional
+            Controls the backend which is combined with the frontend.
+
+        Attributes
+        ----------
+        J : int
+            Log-2 of the scattering scale.
+        shape : tuple of int
+            Spatial support (M, N) of the input.
+        L : int, optional
+            Number of angles used for the wavelet transform.
+        max_order : int, optional
+            The maximum order of scattering coefficients to compute.
+            Must be either equal to `1` or `2`.
+        pre_pad : boolean
+            Controls the padding: if set to False, a symmetric padding is applied
+            on the signal. If set to True, the software will assume the signal was
+            padded externally.
+        Psi : dictionary
+            Contains the wavelets filters at all resolutions. See
+            filter_bank.filter_bank for an exact description.
+        Phi : dictionary
+            Contains the low-pass filters at all resolutions. See
+            filter_bank.filter_bank for an exact description.
+        M_padded, N_padded : int
+             Spatial support of the padded input.
+
+        Notes
+        -----
+        The design of the filters is optimized for the value L = 8.
+
+        pre_pad is particularly useful when cropping bigger images because
+        this does not introduce border effects inherent to padding.
+        """
+        
+    def __init__(self, J, shape, L=8, max_order=2, pre_pad=False,\
+           backend=None):
+        super(Scattering2DBase, self).__init__()
+        self.pre_pad, self.L, self.backend, self.J, self.shape, self.max_order = pre_pad, L, backend, J, shape,\
+                                                                            max_order
+
+    def build(self):
```

Comment:
Add docstring?

Reply:
isn't it done in the inhereited class?

Reply:
I suppose this is true. I guess it seems like we should write down what the errors raised are through, like RuntimeErrors.

Reply:
Go for it! Let's try to avoid redundant documentation